[Zhan Hong] iP

Zhannyhong opened the pull request:
### Gabby
> “I love software engineering ❤️” – Lee Kuan Yew

Gabby frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

___

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public static void main(String[] args) {
    new Gabby("data/tasks.txt").run();
}
```
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Zhannyhong committed:
Rename to Gabby which greets user and exits

Zhannyhong committed:
Refactor greet and bye into functions

Zhannyhong committed:
Echo user commands

Zhannyhong committed:
Add user tasks

Zhannyhong committed:
Display tasks user added

Zhannyhong committed:
Display done status of tasks

Zhannyhong committed:
Add ability to change status of task to done or not done

Zhannyhong committed:
Add Todos, Events, and Deadlines

Zhannyhong committed:
Add command support for tracking Todos, Events, and Deadlines

Zhannyhong committed:
Semi-automate testing

Zhannyhong committed:
Implement basic error handling

Zhannyhong committed:
Add support for deleting tasks from the list

Zhannyhong committed:
Add more personality in responses

Zhannyhong committed:
Update regression testing input and expected output

Zhannyhong committed:
feat: Implement serialization and deserialization of task list

Zhannyhong committed:
feat: Save tasks in the hard disk automatically whenever the task list changes

Zhannyhong committed:
Merge branch 'branch-Level-7'

Zhannyhong committed:
feat: Store deadline and event datetimes as java.time.LocalDateTime objects

Zhannyhong committed:
feat: Check event end time is not before start time

Zhannyhong committed:
feat: List command can filter deadlines occurring on a specific date

Zhannyhong committed:
Merge branch 'branch-Level-8'

Zhannyhong committed:
refactor: Extract classes for Ui, Storage, Parser, TaskList, and Commands

Zhannyhong committed:
refactor: Move logic for validation check of taskID into TaskList.java

Zhannyhong committed:
Merge branch 'branch-A-MoreOOP'

Zhannyhong committed:
refactor: Divide classes into packages

Zhannyhong committed:
Merge branch 'branch-A-Packages'

Zhannyhong committed:
Merge branch 'add-gradle-support'

Zhannyhong committed:
test: Add JUnit tests for Parser and TaskList

Zhannyhong committed:
Merge branch 'branch-A-JUnit'

Zhannyhong committed:
docs: Add JavaDoc comments

Zhannyhong committed:
style: Comply with coding standard

Zhannyhong committed:
feat: Implement find command for users to search for a keyword in task description

Zhannyhong committed:
docs: Add JavaDocs for FindCommand.java

Zhannyhong committed:
Merge branch 'branch-A-JavaDoc'

Zhannyhong committed:
Merge branch 'branch-A-CodingStandard'

Zhannyhong committed:
Merge branch 'branch-Level-9'

Zhannyhong committed:
fix: Add FindCommand to Parser.java imports

Zhannyhong committed:
style: Comply with coding standard for JUnit tests

Zhannyhong committed:
Merge branch 'branch-A-CheckStyle'

Zhannyhong committed:
feat: Add GUI that echoes user input

wei-song1 commented on src/main/java/gabby/Parser.java:
```diff
@@ -0,0 +1,66 @@
+        String command = args[0].toUpperCase();
+        String arg = args.length > 1 ? args[1] : "";
+
+        return switch (command) {
+        case "BYE" -> new ByeCommand();
+        case "LIST" -> new ListCommand(arg);
+        case "MARK" -> new MarkCommand(parseTaskID(arg));
+        case "UNMARK" -> new UnmarkCommand(parseTaskID(arg));
+        case "DELETE" -> new DeleteCommand(parseTaskID(arg));
+        case "TODO" -> new TodoCommand(arg);
+        case "DEADLINE" -> new DeadlineCommand(arg);
+        case "EVENT" -> new EventCommand(arg);
+        case "FIND" -> new FindCommand(arg);
+        default -> throw new GabbyException("Sorry! I don't understand what you just said =(");
+        };
```
According to the coding standard, "Lambda-style switch statements/expressions can have indented case blocks", your code is missing indents

wei-song1 commented on src/main/java/gabby/Gabby.java:
```diff
@@ -0,0 +1,59 @@
+    public void run() {
+        this.ui.showWelcome();
+
+        try (Scanner reader = new Scanner(System.in)) {
+            while (true) {
+                String input = reader.nextLine().strip();
+
+                try {
+                    Command c = Parser.parse(input);
+                    c.execute(this.tasks, this.ui, this.storage);
+
+                    if (c.isExit()) {
+                        break;
+                    }
+                } catch (GabbyException e) {
+                    this.ui.showMsg(e.getMessage());
+                }
+            }
+        }
```
Not sure if it's allowed, there's no catch statement for the first initial try. Coding standard says "A try-catch statement should have the following form" and shows off only try catch and try catch finally

wei-song1 commented on src/main/java/gabby/Parser.java:
```diff
@@ -0,0 +1,66 @@
+     * @throws GabbyException If the input is invalid.
+     */
+    public static Command parse(String input) throws GabbyException {
+        String[] args = input.split(" ", 2);
+        String command = args[0].toUpperCase();
+        String arg = args.length > 1 ? args[1] : "";
```
Maybe rename args and arg to be more definitive? Hard to tell at a glance what is arg and what is args for

wei-song1 commented on src/main/java/gabby/task/TaskList.java:
```diff
@@ -0,0 +1,161 @@
+     */
+    public Task markTask(int taskID) throws GabbyException {
+        if (isTaskIdInvalid(taskID)) {
+            throw new GabbyException("gabby.task.Task ID is not in your list!");
```
I am guessing the gabby.task.Task is not intentional and was renamed by IntelliJ when you were renaming your classes or moving it into packages

wei-song1 reviewed (commented):
Code looks amazing and well maintained, good job. Couldn't really find any mistakes for PR review.

Zhannyhong committed:
feat: Integrate Gabby logic into GUI interface

Zhannyhong committed:
Merge branch 'branch-Level-10'

Engulfy commented on src/main/java/gabby/command/Command.java:
I like how you abstracted out the different commands into different classes

Engulfy commented on src/main/java/gabby/task/EventTask.java:
```diff
@@ -0,0 +1,132 @@
+                    "Datetime provided is in the wrong format. Expected: yyyy-mm-dd hhmm (e.g. 2001-11-23 2025)");
+        }
+
+        if (to.isBefore(from)) {
```
I like how you handled the checking of event date time to ensure the from is before the to

Engulfy commented on src/main/java/gabby/Storage.java:
```diff
@@ -0,0 +1,93 @@
+                    default -> { } // Silently ignore tasks that are not in the correct format
+                    }
+                } catch (GabbyException err) {
+                    // Silently ignore tasks that are not in the correct format
```
Perhaps could give an error message if the tasks were not in the correct format?

Engulfy commented on src/main/java/gabby/Storage.java:
```diff
@@ -0,0 +1,93 @@
+ * Represents a storage class that handles loading and saving of tasks.
+ */
+public class Storage {
+    private String taskStorePath = "./data/tasks.txt";
```
Perhaps can make it final since it will not change?

Engulfy reviewed (commented):
LGTM! Most of the code functionality seems complete and following the coding standards. I like how the Java Docs give good description of the methods as well. Good job!! :D

Zhannyhong committed:
style: Follow Java coding standard

Zhannyhong committed:
test: Add assertions

Current code has no assertions.

Assertions should be used to document important assumptions in the code.

Let's add assertions to ensure that these assumptions hold true during runtime.

Zhannyhong committed:
Merge pull request #1 from Zhannyhong/branch-A-Assertions

test: Add assertions

Zhannyhong committed:
refactor: Improve code quality

Code quality can be improved to increase readability.

Let's,
* Avoid magic literals
* Minimize scope of variables
* Group long blocks of code more logically
* Add fallthrough comment for case statements without a break statement

Zhannyhong committed:
Merge pull request #2 from Zhannyhong/branch-A-CodeQuality

refactor: Improve code quality

Zhannyhong committed:
chore: Set up CI

Zhannyhong committed:
chore: Include checkstyle and suppressions

Zhannyhong committed:
test: Update I/O redirection input and expected output

Zhannyhong committed:
fix: Error in working directory path for Windows

Zhannyhong committed:
feat: Add help command

Zhannyhong committed:
Merge pull request #3 from Zhannyhong/branch-C-Help

feat: Add help command

Zhannyhong committed:
style: Update user and Gabby profile pictures

Zhannyhong committed:
style: Update Gabby format error responses

Zhannyhong committed:
docs: Remove any mentions of Duke

Zhannyhong committed:
Merge branch 'branch-A-Personality'

Zhannyhong committed:
fix: Modify command regexes to accept leading and trailing spaces around parameters

Zhannyhong committed:
fix: Use strict resolver style for datetimes

Zhannyhong committed:
fix: Modify datetime format to work with strict resolver

Zhannyhong committed:
Merge branch 'branch-A-MoreErrorHandling'

Zhannyhong committed:
docs: Add user guide

Zhannyhong committed:
Merge branch 'branch-A-UserGuide'

Zhannyhong committed:
fix: Escape delimiters in task descriptions

Zhannyhong committed:
style: Modify lambda-style switch statements to have indentations
